src/pages/ContentfulPage.vue

```diff
@@ -304,35 +304,37 @@ export default {
 				}
 			}
 
-			return client.query({
+			const { data } = await client.query({
 				query: contentfulEntries,
 				variables: {
 					contentType: 'page',
 					contentKey: args?.route?.meta?.contentfulPage(
 						args?.route, client, experimentVersionFragment
 					)?.trim(),
 				}
-			}).then(({ data }) => {
-				// Get Contentful page data
-				const pageData = getPageData(data);
-				if (pageData.error) {
-					// Only import the error page if there is a contentful error
-					return Promise.all([ErrorPage()]);
-				}
-				// Get page frame component
-				const pageFrame = getPageFrameFromType(pageData?.page?.pageType);
-				// Get components for content groups
-				const contentGroups = getContentGroups(pageData);
-				// Start importing all components
-				return Promise.all([
-					pageFrame(),
-					...contentGroups.map(g => g.component()),
-				]);
-			}).then(resolvedImports => {
-				// Call preFetch for page frame and content group components
-				const components = resolvedImports.map(resolvedImport => resolvedImport.default);
-				return preFetchAll(components, client, args);
 			});
+			// Get Contentful page data
+			const pageData = getPageData(data);
+			if (pageData.error) {
+				// Only prefetch the error page if there is a contentful error
+				return preFetchAll([ErrorPage], client, args);
+			}
+			// Get page frame component
+			const pageFrame = getPageFrameFromType(pageData?.page?.pageType);
+			// Get components for content groups
+			const contentGroups = getContentGroups(pageData);
+			// Call preFetch for page frame and content group components
+			return Promise.all([
+				preFetchAll([pageFrame], client, args),
+				...contentGroups.map(group => preFetchAll(
+					[group.component],
+					client,
+					{
+						...args,
+						content: group.content,
+					}
+				))
+			]);
```

Comment:
Just a little re-arranging! Thanks for keeping this separate from any content group's internal concerns.